packages/swing-store/src/snapStore.js

```diff
@@ -338,21 +388,34 @@ export function makeSnapStore(
     WHERE vatID = ?
   `);
 
+  const sqlGetSnapshotList = db.prepare(`
+    SELECT endPos
+    FROM snapshots
+    WHERE vatID = ?
+    ORDER BY endPos
+  `);
+  sqlGetSnapshotList.pluck(true);
+
   /**
    * Delete all snapshots for a given vat (for use when, e.g., a vat is terminated)
    *
    * @param {string} vatID
    */
   function deleteVatSnapshots(vatID) {
+    ensureTxn();
+    const deletions = sqlGetSnapshotList.all(vatID);
+    for (const endPos of deletions) {
+      const exportRec = snapshotRec(vatID, endPos, undefined);
```

Comment:
nit

While there is no type check, let's assume there was.

```suggestion
      const exportRec = snapshotRec(vatID, endPos, '', 0);
```

Reply:
I actually want the corresponding properties of the resulting object to be absent from it, so I think the correct handling of this is to add the JSDoc typedefs and make those last two parameters explicitly optional.